frontend/src/app/main/ui/workspace/sidebar/layers.cljs

```diff
@@ -229,156 +249,162 @@
-                   :data-filter "component"
-                   :on-click add-filter}
-              [:div {:class (stl/css :filter-menu-item-name-wrapper)}
-               [:span {:class (stl/css :filter-menu-item-icon)}
-                i/component-refactor]
-               [:span {:class (stl/css :filter-menu-item-name)}
-                (tr "workspace.sidebar.layers.components")]]
-
-              (when (contains? current-filters :component)
-                [:span {:class (stl/css :filter-menu-item-tick)}
-                 i/tick-refactor])]
-
-             [:li {:class (stl/css-case :filter-menu-item true
-                                        :selected (contains? current-filters :text))
-                   :data-filter "text"
-                   :on-click add-filter}
-              [:div {:class (stl/css :filter-menu-item-name-wrapper)}
-               [:span {:class (stl/css :filter-menu-item-icon)}
-                i/text-refactor]
-               [:span {:class (stl/css :filter-menu-item-name)}
-                (tr "workspace.sidebar.layers.texts")]]
-
-              (when (contains? current-filters :text)
-                [:span {:class (stl/css :filter-menu-item-tick)}
-                 i/tick-refactor])]
-
-             [:li {:class (stl/css-case :filter-menu-item true
-                                        :selected (contains? current-filters :image))
-                   :data-filter "image"
-                   :on-click add-filter}
-              [:div {:class (stl/css :filter-menu-item-name-wrapper)}
-               [:span {:class (stl/css :filter-menu-item-icon)}
-                i/img-refactor]
-               [:span {:class (stl/css :filter-menu-item-name)}
-                (tr "workspace.sidebar.layers.images")]]
-
-              (when (contains? current-filters :image)
-                [:span {:class (stl/css :filter-menu-item-tick)}
-                 i/tick-refactor])]
-
-             [:li {:class (stl/css-case :filter-menu-item true
-                                        :selected (contains? current-filters :shape))
-                   :data-filter "shape"
-                   :on-click add-filter}
-              [:div {:class (stl/css :filter-menu-item-name-wrapper)}
-               [:span {:class (stl/css :filter-menu-item-icon)}
-                i/path-refactor]
-               [:span {:class (stl/css :filter-menu-item-name)}
-                (tr "workspace.sidebar.layers.shapes")]]
-
-              (when (contains? current-filters :shape)
-                [:span {:class (stl/css :filter-menu-item-tick)}
-                 i/tick-refactor])]])]
-
-         [:div {:class (stl/css :tool-window-bar)}
-          [:& title-bar {:collapsable  false
-                         :title        (:name page)
-                         :on-btn-click toggle-search
-                         :btn-children i/search-refactor}]]))]))
+       (do
+         (mf/with-effect
```

Comment:
I don't think is a good idea setting up the effect here, It's probably better to setting up the effect hook before returning the vector (line 258 per example)